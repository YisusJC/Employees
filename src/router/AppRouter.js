import React from "react";
import { Routes, Route } from "react-router-dom";
import { AddEmployee } from "../components/AddEmployee/AddEmployee";
import { EditEmployee } from "../components/EditEmployee/EditEmployee";
import { Employees } from "../components/Employees/Employees";

export const AppRouter = () => {
  return (
    <div>
      <Routes>
        <Route exact path="/add-employee" element={<AddEmployee />} />
        <Route exact path="/edit-employee/:id" element={<EditEmployee />} />
        <Route exact path="/search/:sp" element={<Employees />} />
        <Route exact path="/" element={<Employees />} />
      </Routes>
    </div>
  );
};
