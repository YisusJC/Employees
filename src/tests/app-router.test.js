import { mount, shallow } from "enzyme/build";
import { AppRouter } from "../router/AppRouter";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

// App Router Jest tests
describe("App Router Component", () => {
  // should render without crashing.
  it("should render without crashing", () => {
    const wrapper = shallow(<AppRouter />);
    expect(wrapper).toMatchSnapshot();
  });
});
