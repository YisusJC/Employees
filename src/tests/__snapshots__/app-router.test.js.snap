// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Router Component should render without crashing 1`] = `ShallowWrapper {}`;
